test/python/opflow/test_gradients.py

```diff
@@ -876,6 +876,50 @@ def test_gradient_wrapper(self, backend_type):
             self.assertTrue(np.allclose(result[0], correct_values[i][0], atol=0.1))
             self.assertTrue(np.allclose(result[1], correct_values[i][1], atol=0.1))
 
+    def test_gradient_wrapper2(self):
+        """Test the gradient wrapper for gradients checking that statevector and qasm gives the
+           same results
+
+        dp0/da = cos(a)sin(b) / 2
+        dp1/da = - cos(a)sin(b) / 2
+        dp0/db = sin(a)cos(b) / 2
+        dp1/db = - sin(a)cos(b) / 2
+        """
+        method = 'lin_comb'
+        a = Parameter('a')
+        b = Parameter('b')
+        params = [a, b]
+        backends = ['statevector_simulator', 'qasm_simulator']
+
+        qc = QuantumCircuit(2)
+        qc.h(1)
+        qc.h(0)
+        qc.sdg(1)
+        qc.cz(0, 1)
+        qc.ry(params[0], 0)
+        qc.rz(params[1], 0)
+        qc.h(1)
+
+        obs = (Z ^ I) - 1j * (Y ^ I)
+        op = ~StateFn(obs) @ CircuitStateFn(primitive=qc, coeff=1.)
+
+        shots = 8000
+
+        values = [[0, np.pi / 2], [np.pi / 4, np.pi / 4], [np.pi / 3, np.pi / 9]]
+        for i, value in enumerate(values):
+            results = []
+            for backend_type in backends:
+                backend = BasicAer.get_backend(backend_type)
+
+                q_instance = QuantumInstance(backend=backend, shots=shots)
+
+                grad = NaturalGradient(grad_method=method).gradient_wrapper(operator=op,
+                                                                     bind_params=params,
+                                                                     backend=q_instance)
```

Comment:
Pylint complains about the indentation, this should fix it 🙂 
```suggestion
                grad = NaturalGradient(grad_method=method).gradient_wrapper(
                    operator=op, bind_params=params, backend=q_instance
                )
```